neuromation/api/images.py

```diff
@@ -140,20 +144,65 @@ def _auth(self) -> Dict[str, str]:
             raise  # pragma: no cover
 
         async for obj in stream:
-            if "error" in obj.keys():
-                error_details = obj.get("errorDetail", {"message": "Unknown error"})
-                raise DockerError(900, error_details)
-            elif "id" in obj.keys() and obj["id"] != remote.tag:
-                if "progress" in obj.keys():
-                    message = f"{obj['id']}: {obj['status']} {obj['progress']}"
-                else:
-                    message = f"{obj['id']}: {obj['status']}"
-                progress.step(ImageProgressStep(message, obj["id"]))
+            step = _try_parse_image_progress_step(obj, remote.tag)
+            if step:
+                progress.step(step)
 
         await self._docker.images.tag(repo, local)
 
         return local
 
+    async def save(
+        self,
+        id: str,
+        image: RemoteImage,
+        *,
+        container_progress: Optional[AbstractDockerContainerProgress] = None,
+        image_progress: Optional[AbstractDockerImageProgress] = None,
+    ) -> None:
+        if not _is_in_neuro_registry(image):
+            raise ValueError(f"Image `{image}` must be in the neuromation registry")
+        if container_progress is None:
+            container_progress = _DummyContainerProgress()
+        if image_progress is None:
+            image_progress = _DummyImageProgress()
+
+        payload = {"container": {"image": _as_repo_str(image)}}
+        url = self._config.cluster_config.monitoring_url / f"{id}/save"
+
+        async with self._core.request("POST", url, json=payload) as resp:
+
+            # first, we process docker-commit messages
+            container_progress.commit(ContainerProgressCommit(id, image))
+            async for chunk in resp.content:
+                obj = self._load_chunk(chunk)
+                commit_step = _try_parse_job_commit_progress_step(obj)
```

Comment:
AFAIK, docker-commit does not support streaming (even though it can take a while, for larger images). Thus, there always will be only 2 messages, "about to commit" and "committed", or an error message. So let's not print these messages at all, and just check if there's no errors?

Reply:
the code will be simplified significantly

Reply:
Agree, the commit progress is pointless if docker doesn't provide *progress* information for it.